File: tmp_diagnose.py
from pathlib import Path
from babel import parser
from babel import traverse

source = Path('src/screens/HomeScreen.js').read_text('utf-8')
try:
    ast = parser.parse(source, source_type='module', plugins=['jsx','classProperties'])
    print('parsed ok')
except Exception as e:
    print(type(e).__name__, e)
    if hasattr(e, 'loc'):
        print('loc', e.loc)

File: tmp_diagnose.js
const fs = require("fs");
const parser = require("@babel/parser");
const source = fs.readFileSync("src/screens/HomeScreen.js", "utf8");
try {
  parser.parse(source, {
    sourceType: "module",
    plugins: ["jsx", "classProperties"],
  });
  console.log("parsed OK");
} catch (e) {
  console.error(e.message);
  if (e.loc) console.error("loc", e.loc);
  if (e.codeFrame) console.error(e.codeFrame);
}
